### ISSUE TITLE
Circular reference exception sometimes thrown incorrectly for SUMIF, COUNTIFS

### ISSUE THREAD

**Issue by craigbrown** [EPPlusSoftware/EPPlus#635]
If I have this formula in cell `A2`:
```
// Sum every cell in row 2 that has the string "SUMMABLE" in the cell above it.
=SUMIF(1:1,"SUMMABLE",2:2)
```
then Excel will _not_ recognize this as a circular reference unless `A1` has the value "SUMMABLE". Although `A2` is part of the range `2:2`, Excel calculates the filtered set of cells first, and decides whether it's a circular reference based on that.

![SumIfSameRowExample](https://user-images.githubusercontent.com/8934235/166569087-ccef44f6-eba1-4749-9a04-4d17969e2a7a.png)

EPPlus sees this cell as a circular reference, so won't try to calculate its value. Is there any way circular references in EPPlus could be improved to handle this case?

I've included some unit tests below, including one for the `COUNTIFS` function which is also affected. 

Many thanks as always!

```csharp
[TestMethod]
public void SumIf_SumThisRowWithoutCircularReferences()
{
    using (var pck = new ExcelPackage())
    {
        var sheet1 = pck.Workbook.Worksheets.Add("Sheet1");
        sheet1.Cells["A1"].Value = "SumResult";
        // This shouldn't be a circular reference, because the 1:1="SUMMABLE" condition should filter out A2
        sheet1.Cells["A2"].Formula = "SUMIF(1:1,\"SUMMABLE\",2:2)";
        
        sheet1.Cells["B2"].Value = 1;
        sheet1.Cells["C2"].Value = 2;
        sheet1.Cells["D2"].Value = 3;
        sheet1.Cells["E2"].Value = 4;
        sheet1.Cells["F2"].Value = 5;
        sheet1.Cells["G2"].Value = 6;

        sheet1.Cells["C1"].Value = "SUMMABLE";
        sheet1.Cells["D1"].Value = "SUMMABLE";
        sheet1.Cells["E1"].Value = "SUMMABLE";
        sheet1.Cells["G1"].Value = "SUMMABLE";

        pck.Workbook.Calculate();

        Assert.AreEqual(15, sheet1.Cells["A2"].GetValue<double>(), Tolerance);
    }
}
[TestMethod]
public void CountIfs_CountThisRowWithoutCircularReferences()
{
    using (var pck = new ExcelPackage())
    {
        var sheet1 = pck.Workbook.Worksheets.Add("Sheet1");
        sheet1.Cells["A1"].Value = "SumResult";
        // This shouldn't be a circular reference, because the 1:1="COUNTABLE" condition should filter out A2 before the 2:2 filter is applied
        sheet1.Cells["A2"].Formula = "COUNTIFS(1:1,\"COUNTABLE\",2:2,\"<>\")";

        sheet1.Cells["B2"].Value = 1;
        sheet1.Cells["C2"].Value = 2;
        sheet1.Cells["E2"].Value = 4;
        sheet1.Cells["F2"].Value = 5;
        sheet1.Cells["G2"].Value = 6;

        sheet1.Cells["C1"].Value = "COUNTABLE";
        sheet1.Cells["D1"].Value = "COUNTABLE";
        sheet1.Cells["E1"].Value = "COUNTABLE";
        sheet1.Cells["G1"].Value = "COUNTABLE";

        pck.Workbook.Calculate();

        Assert.AreEqual(3, sheet1.Cells["A2"].GetValue<double>(), Tolerance);
    }
}
```

**Comment by swmal** [comment]
Thanks Craig, we'll have a look at this.

**Comment by swmal** [comment]
I should just mention that this will work out of the box if you call Calculate like this:

`pck.Workbook.Calculate(x => x.AllowCircularReferences = true);`

You can also set this in your config file, see this page: https://github.com/EPPlusSoftware/EPPlus/wiki/Circular-references

I will try to fix this issue anyway we have already done a similar solution for the IF function.

**Comment by craigbrown** [comment]
> I should just mention that this will work out of the box if you call Calculate like this:
> 
> `pck.Workbook.Calculate(x => x.AllowCircularReferences = true);`

That does stop the circular reference exception from being thrown, but the value of the `SUMIF`/`COUNTIF` formula is just calculated as `0` (in v6.0.3 anyway).

Thanks for looking into a fix for this!

**Comment by craigbrown** [comment]
Just tested this and it is working now. Thanks very much!

**Comment by JanKallman** [comment]
Fixed in version 5.8.10 and 6.0.4
